### PR TITLE
Issue/3204 project install tests

### LINKED ISSUE #3204: modules v2 pypi loader test: expand project install tests
Expand `tests/moduletool/test_install::test_project_install` to handle more edge cases:
- v2 already installed doesn't get overridden
- all failure paths reachable by `Project.verify()` (called before plugin loading) and `Project.verify_python_environment()` called after plugin loading), for inter-v1, inter-v2 and v1-to-v2 dependencies.
- ...?

### PR DESCRIPTION
**Pull request by sanderr**
# Description

- expanded tests for project install
- added support for dev tagged versions for v2 modules

This PR uses the `setup_simple_project` helper function in tests while concurrent work expanded the snippetcompiler fixture to support required behavior. #3235 has been created to fix this.

Closes #3204

# Self Check:

Strike through any lines that are not applicable (`~~line~~`) then check the box

- [x] Attached issue to pull request
- [x] Changelog entry
- [x] Type annotations are present
- [x] Code is clear and sufficiently documented
- [x] No (preventable) type errors (check using make mypy or make mypy-diff)
- [x] Sufficient test cases (reproduces the bug/tests the requested feature)
- [x] Correct, in line with design
- [x] ~~End user documentation is included or an issue is created for end-user documentation (add ref to issue here: )~~

# Reviewer Checklist:

- [ ] Sufficient test cases (reproduces the bug/tests the requested feature)
- [ ] Code is clear and sufficiently documented
- [ ] Correct, in line with design

### CODE REVIEW

File: tests/moduletool/test_install.py [hunk 1]
```diff
@@ -16,44 +16,161 @@
...
+    if install:
+        ModuleTool().install(editable=editable, path=dest_dir)
+    if publish_index is not None:
+        ModuleTool().build(path=dest_dir, output_dir=publish_index.artifact_dir)
+        publish_index.publish()
+    with open(config_file, "r") as fh:
+        return module.ModuleV2Metadata.parse(fh)
+
+
+def setup_simple_project(
```
Comment (sanderr) [inline tests/moduletool/test_install.py:142]: I just moved this up in the file.

File: tests/moduletool/test_install.py [hunk 2]
```diff
@@ -16,44 +16,161 @@
...
 
 
-def run_module_install(python_path: str, module_path: str, editable: bool, set_path_argument: bool) -> None:
+@pytest.fixture
+def build_venv_active(tmpvenv_active: Tuple[py.path.local, py.path.local]) -> Iterator[Tuple[py.path.local, py.path.local]]:
     """
-    Install the Inmanta module (v2) using the `inmanta module install` command.
+    Yields an active virtual environment that is suitable to build modules with.
+    """
+    env.process_env.install_from_index([Requirement.parse("virtualenv")])
```
Comment (wouterdb) [inline tests/moduletool/test_install.py:50]: do we still use `virtualenv`? I assumed we where on `venv` completely? 

(I am quite certain we should be, or all hell would break loose)

**Review by wouterdb — CHANGES_REQUESTED** [review]

File: tests/moduletool/test_install.py (hunk 2 above)
Comment (sanderr) [reply tests/moduletool/test_install.py:50]: It's the `build` package that requires it. Installing `build` instead seems to do the trick as well, that's cleaner.

**Review by arnaudsjs — CHANGES_REQUESTED** [review]

File: tests/conftest.py [hunk 3]
```diff
@@ -789,37 +789,40 @@ def setup_for_snippet(
...
         """
-        assert project.virtualenv.virtual_python is not None
-        env.process_env.__init__(python_path=project.virtualenv.virtual_python)
+        env.process_env.__init__(env_path=self.env)
         path: str = os.path.join(env.process_env.site_packages_dir, const.PLUGINS_PACKAGE)
         env.process_env.notify_change()
         os.makedirs(path, exist_ok=True)
 
-    def _install_v2_modules(self, project: Project, install_v2_modules: List[LocalPackagePath] = []) -> None:
+    def _install_v2_modules(self, project: Project, install_v2_modules: List[LocalPackagePath] = None) -> None:
```
[the remark this answers is not in the document]
Comment (sanderr) [reply tests/conftest.py:824]: Thanks

**Review by arnaudsjs — APPROVED** [review]

File: tests/moduletool/test_install.py (hunk 2 above)
Comment (wouterdb) [reply tests/moduletool/test_install.py]: scared me there

**Review by wouterdb — APPROVED** [review]
